Added funcationality

MLBZ521 opened the pull request:
This builds on #53 and #54.

Refactored code to hopefully make it easier to understand what's going on (but maybe not? 😁 ).

Also, for what I was planning to use this for, I needed additional JSSImporter details.  I don't think this would break any current uses of the processor (besides what #54 changes).

However, just incase you don't want to merge this (or other PRs), I created separate PRs so you can decide.  And if not, I'll create a forked version in my own repo.

grahampugh commented:
Hi @MLBZ521 , sorry for not getting back to these for a long time. I think I need a primer on what all the three different PRs are for, and how they would work in combination, i.e. in which order they would need to be merged.

MLBZ521 commented:
Hey @grahampugh, no worries.  I had actually updated this processor _again_ the day before your message, but wanted to test it further before including that in this PR -- which I just did.

If you wanted to include them all, they should be merged in the order of the PR.

Regarding what I changed, do you want to review that here or discuss over on the MacAdmins Slack?  In short, nothing should _**really**_ change the end result (I don't think), unless someone didn't want _**all**_ the info I've updated it to grab now.  The only _breaking change_ that I'm aware of would be changing the processors' input variable `RECIPE_NAME` to `recipe_id`.  This is to allow supplying full recipe identifiers in case recipe identifiers are not in the format of `local.jss.<NAME>`.

By the way, your processor is a brilliant idea and exactly what I was looking to do from my development -> to production runs in the automated workflows I've setup in my environment.

grahampugh commented:
Thanks for the reminder. I'd kind of forgotten what this was about :)

I actually use an internal version of this which has a couple of improvements in it that I should try and merge in too - mainly it now deletes old receipts to prevent the search process from taking too long as the receipts build up over time.

So as soon as I get a chance, I'll make a new branch, merge in my changes and then take it from there.

Fin case you're interested, I have started moving to a different set of processors for this purpose, where instead of relying on the receipts, I write a JSON file out in the Cache folder with the values I need. This is a bit faster than using JSSRecipeReceiptsChecker, but requires `com.github.grahampugh.recipes.postprocessors/LastRecipeRunResult` at the end of the "development" jamf recipe, and `com.github.grahampugh.recipes.preprocessors/LastRecipeRunChecker` as the first process in the "production" recipe. 

For examples, see https://github.com/eth-its/autopkg-mac-recipes-yaml/tree/main/Jamf_Recipes` for the "Development" recipes, and https://github.com/eth-its/autopkg-mac-recipes-yaml/tree/main/Jamf_Staging_Recipes for the "Production" recipes.

MLBZ521 commented:
Hmmm...  Sounds interesting, I'll have to take a look.  Though...I'm literally just wrapping up an automation process (with a Slack Bot) that controls all this using `JSSRecipeReceiptsChecker`.  I spent quite a bit of time going back and forth on how to structure all this between the end of 2019 and beginning of 2020.  Finally got to work on putting the ideas to code earlier this year and have it running in production.  Just making some touch ups to account for any non-standard recipe runs, etc.
